sdk/monitor/Azure.Monitor.OpenTelemetry.LiveMetrics/src/Internals/DocumentHelper.cs

```diff
@@ -0,0 +1,161 @@
+                                                ? activity.Duration.ToString("c", CultureInfo.InvariantCulture)
+                                                : SchemaConstants.Duration_MaxValue,
+                // TODO: Properties = new Dictionary<string, string>(), - UX supports up to 10 custom properties
+
+                // The following properties are used to calculate metrics. These are not serialized.
+                Extension_IsSuccess = IsSuccess(activity, httpResponseStatusCode),
+                Extension_Duration = activity.Duration.TotalMilliseconds,
+            };
+
+            return remoteDependencyDocumentIngress;
+        }
+
+        internal static Request ConvertToRequest(Activity activity)
+        {
+            string httpResponseStatusCode = string.Empty;
+            string urlScheme = string.Empty;
+            string serverAddress = string.Empty;
+            string serverPort = string.Empty;
+            string urlPath = string.Empty;
+            string urlQuery = string.Empty;
+
+            foreach (ref readonly var tag in activity.EnumerateTagObjects())
+            {
+                if (tag.Value == null)
+                {
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlScheme)
+                {
+                    urlScheme = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeServerAddress)
+                {
+                    serverAddress = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeServerPort)
+                {
+                    serverPort = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlPath)
+                {
+                    urlPath = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeUrlQuery)
+                {
+                    urlQuery = tag.Value.ToString();
+                    continue;
+                }
+                else if (tag.Key == SemanticConventions.AttributeHttpResponseStatusCode)
+                {
+                    httpResponseStatusCode = tag.Value.ToString();
+                    continue;
+                }
+            }
+
+            var length = urlScheme.Length + Uri.SchemeDelimiter.Length + serverAddress.Length + serverPort.Length + urlPath.Length + urlQuery.Length;
+            var urlStringBuilder = new System.Text.StringBuilder(length)
+                .Append(urlScheme)
+                .Append(Uri.SchemeDelimiter)
+                .Append(serverAddress)
+                .Append(serverPort != null ? $":{serverPort}" : string.Empty)
+                .Append(urlPath)
+                .Append(urlQuery);
```

Comment:
why not build it to string here?

Reply:
I don't need the extra variable so calling ToString inline below.